src/data/polynomial/degree/definitions.lean

```diff
@@ -727,23 +730,91 @@ by erw [degree_mul' this, degree_eq_nat_degree hp,
 
-lemma ne_zero_of_coe_le_degree {P : polynomial R} {n : ℕ} (hdeg : ↑n ≤ P.degree) : P ≠ 0 :=
+lemma ne_zero_of_coe_le_degree (hdeg : ↑n ≤ p.degree) : p ≠ 0 :=
 by rw ← degree_nonneg_iff_ne_zero; exact trans (by exact_mod_cast n.zero_le) hdeg
 
-lemma le_nat_degree_of_coe_le_degree {P : polynomial R} {n : ℕ} (hdeg : ↑n ≤ P.degree) :
-  n ≤ P.nat_degree :=
+lemma le_nat_degree_of_coe_le_degree (hdeg : ↑n ≤ p.degree) :
+  n ≤ p.nat_degree :=
 with_bot.coe_le_coe.mp ((degree_eq_nat_degree $ ne_zero_of_coe_le_degree hdeg) ▸ hdeg)
 
+lemma nat_degree_le_iff_coeff_eq_zero :
+  p.nat_degree ≤ n ↔ ∀ N : ℕ, n < N → p.coeff N = 0 :=
+by simp_rw [nat_degree_le_iff_degree_le, degree_le_iff_coeff_zero, with_bot.coe_lt_coe]
+
+lemma nat_degree_C_mul_le (a : R) (f : polynomial R) :
+  (C a * f).nat_degree ≤ f.nat_degree :=
+calc
+  (C a * f).nat_degree ≤ (C a).nat_degree + f.nat_degree : nat_degree_mul_le
+  ... = 0 + f.nat_degree : by rw nat_degree_C a
+  ... = f.nat_degree : zero_add _
+
+lemma nat_degree_mul_C_le (f : polynomial R) (a : R) :
+  (f * C a).nat_degree ≤ f.nat_degree :=
+calc
+  (f * C a).nat_degree ≤ f.nat_degree + (C a).nat_degree : nat_degree_mul_le
+  ... = f.nat_degree + 0 : by rw nat_degree_C a
+  ... = f.nat_degree : add_zero _
+
+lemma eq_nat_degree_of_le_mem_support (pn : p.nat_degree ≤ n) (ns : n ∈ p.support) :
+  p.nat_degree = n :=
+le_antisymm pn (le_nat_degree_of_mem_supp _ ns)
+
+lemma nat_degree_C_mul_eq_of_mul_eq_one {ai : R} (au : ai * a = 1) :
+  (C a * p).nat_degree = p.nat_degree :=
+le_antisymm (nat_degree_C_mul_le a p) (calc
+  p.nat_degree = (1 * p).nat_degree : by nth_rewrite 0 [← one_mul p]
+  ... = (C ai * (C a * p)).nat_degree : by rw [← C_1, ← au, ring_hom.map_mul, ← mul_assoc]
+  ... ≤ (C a * p).nat_degree : nat_degree_C_mul_le ai (C a * p))
+
+lemma nat_degree_mul_C_eq_of_mul_eq_one {ai : R} (au : a * ai = 1) :
+  (p * C a).nat_degree = p.nat_degree :=
+le_antisymm (nat_degree_mul_C_le p a) (calc
+  p.nat_degree = (p * 1).nat_degree : by nth_rewrite 0 [← mul_one p]
+  ... = ((p * C a) * C ai).nat_degree : by rw [← C_1, ← au, ring_hom.map_mul, ← mul_assoc]
+  ... ≤ (p * C a).nat_degree : nat_degree_mul_C_le (p * C a) ai)
+
+/-- Although not explicitly stated, the assumptions of this lemma force the polynomial `p` to be
+non-zero, via `p.leading_coeff ≠ 0`.  The lemmas below split cases, effectively removing this
+assumption in some cases.
+-/
+lemma nat_degree_mul_C_eq_of_mul_ne_zero (h : p.leading_coeff * a ≠ 0) :
+  (p * C a).nat_degree = p.nat_degree :=
+begin
+  refine eq_nat_degree_of_le_mem_support (nat_degree_mul_C_le p a) _,
+  refine mem_support_iff_coeff_ne_zero.mpr _,
+  rwa coeff_mul_C,
+end
+
+/-- Although not explicitly stated, the assumptions of this lemma force the polynomial `p` to be
+non-zero, via `p.leading_coeff ≠ 0`.  The lemmas below split cases, effectively removing this
+assumption in some cases.
+-/
+lemma nat_degree_C_mul_eq_of_mul_ne_zero (h : a * p.leading_coeff ≠ 0) :
```

Comment:
Can you explicitly name the lemmas you're referring to here, so that they link in doc-gen?

Reply:
Done!  Thanks for the suggestion!